### PR TITLE
[SPARK-46515] Add MONTHNAME function

### PR DESCRIPTION
**Pull request by stefankandic**
### What changes were proposed in this pull request?

Added MONTHNAME function which returns three-letter abbreviated month name for a given date to:
- Scala API
- Python API
- R API
- Spark Connect Scala Client
- Spark Connect Python Client

### Why are the changes needed?

for parity with Snowflake

### Does this PR introduce _any_ user-facing change?

Yes, new MONTHNAME function

### How was this patch tested?

With newly added unit tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -1091,6 +1091,20 @@ setMethod("dayofyear",
             column(jc)
           })
 
+#' @details
+#' \code{monthname}: Extracts the three-letter abbreviated month name from a
+#' given date/timestamp/string.
+#'
+#' @rdname column_datetime_functions
+#' @aliases monthname monthname,Column-method
+#' @note monthname since 3.5.0
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R:1100]: should be 4.0.0

Comment (stefankandic) [reply R/pkg/R/functions.R:1100]: thanks! will change everywhere

**Comment by srielau** [comment]
Parity with what?
When I look at mySQL:
https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_monthname
It seems to print out the complete name. It also seems to obey the locale.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala [hunk 2]
```diff
@@ -911,6 +911,25 @@ case class WeekOfYear(child: Expression) extends GetDateField {
     copy(child = newChild)
 }
 
+// scalastyle:off line.size.limit
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala:914]: Is it really needed?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala [hunk 3]
```diff
@@ -196,6 +197,17 @@ object DateTimeUtils extends SparkDateTimeUtils {
     localDateToDays(daysToLocalDate(days).plusMonths(months))
   }
 
+  /**
+   * Returns the three-letter abbreviated month name for the given number of days 1970-01-01.
+   */
+  def getMonthName(days: Int): UTF8String = {
+    val monthName = Month
+      .of(getMonth(days))
+      .getDisplayName(TextStyle.SHORT, Locale.US)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala:206]: Could you use `DateFormatter.defaultLocale`, please. (at least we don't spread the default local settings across source code).

Comment (stefankandic) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala]: makes sense!

**Review by srielau — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @stefankandic and @srielau for review.

**Comment by stefankandic** [comment]
Thanks for the review! I guess it only makes sense to also add support for DAYNAME for consistency purposes? @srielau @MaxGekk

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala [hunk 4]
```diff
@@ -911,6 +911,23 @@ case class WeekOfYear(child: Expression) extends GetDateField {
     copy(child = newChild)
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(date) - Returns the three-letter abbreviated month name from the given date.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('2008-02-20');
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala:918]: in the function doc, shall we avoid implicit cast and use `DATE'2008-02-20'`?

**Review by beliefer — COMMENTED** [review]
Late LGTM.

**Pull request closed without merging**